src/WebJobs.Extensions.OpenAI.CosmosDBSearch/CosmosDBSearchProvider.cs

```diff
@@ -0,0 +1,271 @@
+        MongoClient searchIndexClient = await this.GetMongoClientAsync(connectionString);
+
+        try
+        {
+            IMongoCollection<BsonDocument> collection = searchIndexClient.GetDatabase(databaseName).GetCollection<BsonDocument>(request.ConnectionInfo.CollectionName ?? defaultSearchIndexName);
+
+            // Search Azure Cosmos DB for MongoDB vCore collection for similar embeddings and project fields.
+            BsonDocument[] pipeline = new BsonDocument[]
+            {
+                new BsonDocument
+                {
+                    {
+                        "$search", new BsonDocument
+                        {
+                            {
+                                "cosmosSearch", new BsonDocument
+                                {
+                                    { "vector", !(request.Embeddings.IsEmpty) ? new BsonArray(request.Embeddings.ToArray()) : new BsonArray()},
+                                    { "path", "embedding" },
+                                    { "k", this.isSemanticSearchEnabled && !request.Embeddings.IsEmpty ? Math.Max(50, request.MaxResults) : request.MaxResults }
+                                }
+                            },
+                            { "returnStoredSource", true }
+                        }
+                    }
+                },
+                new BsonDocument
+                {
+                    {
+                        "$project", new BsonDocument
+                        {
+                            { "embedding", 0 },
+                            { "_id", 0 },
+                            { "id", 0 },
+                            { "timestamp", 0 }
+                        }
+                    }
+                }
+            };
+
+            IAsyncCursor<BsonDocument> cursor = await collection.AggregateAsync<BsonDocument>(pipeline);
+            List<BsonDocument> documents = await cursor.ToListAsync();
+
+            var resultFromDocuments = documents.ToList().ConvertAll(bsonDocument => BsonSerializer.Deserialize<OutputDocument>(bsonDocument));
+
+            List<SearchResult> searchResults = new();
+            
+            foreach (OutputDocument doc in resultFromDocuments)
+            {
+                searchResults.Add(new SearchResult(doc.Title, doc.Text));
+            }
+
+            SearchResponse response =  new SearchResponse(searchResults);
+            return response;
+
+        }
+        catch (MongoException ex)
+        {
+            this.logger.LogError($"Exception: SearchAsync(): {ex.Message}");
+            throw;
+        }
+    }
+
+    void GetAzureAISearchConfig(IOptions<CosmosDBSearchConfigOptions> cosmosDBSearchConfigOptions)
+    {
+        this.isSemanticSearchEnabled = cosmosDBSearchConfigOptions.Value.IsSemanticSearchEnabled;
+        int value = cosmosDBSearchConfigOptions.Value.VectorSearchDimensions;
+        if (value < 2 || value > 2000)
```

Comment:
we should move this to constants at the top of this file, will be easier to find and update in future if required. Feel free to suggest another way for future maintenance.  You can make the same change for AI Search too.

Reply:
I made the values readonly and modified them in the constructor. Let me know if this look good